lib/srv/app/session.go

```diff
@@ -237,37 +282,32 @@ func (s *sessionCache) get(key string) (*session, error) {
 }
 
 // set will add the session to the cache.
-func (s *sessionCache) set(key string, value *session, ttl time.Duration) error {
+func (s *sessionCache) set(value *session, ttl time.Duration) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	if err := s.cache.Set(key, value, ttl); err != nil {
+	if err := s.cache.Set(value.id, value, ttl); err != nil {
 		return trace.Wrap(err)
 	}
 	return nil
 }
 
 // expire will close the stream writer.
 func (s *sessionCache) expire(key string, el interface{}) {
-	session, ok := el.(*session)
-	if !ok {
-		s.log.Debugf("Invalid type stored in cache: %T.", el)
-		return
-	}
-
-	// Closing the stream writer may trigger a flush operation which could be
+	// Closing the session stream writer may trigger a flush operation which could be
 	// time-consuming. Launch in another goroutine since this occurs under a
 	// lock and expire can get called during a "get" operation on the ttlmap.
```

Comment:
nit: respect current line breaks? (Or, alternatively, "disrespect" them consistently?)
```suggestion
	// Closing the session stream writer may trigger a flush operation which could
	// be time-consuming. Launch in another goroutine since this occurs under a
	// lock and expire can get called during a "get" operation on the ttlmap.
```